src/groupeddataframe/splitapplycombine.jl

```diff
@@ -201,6 +201,7 @@ which is determined using the following rules:
   for each group as the length of the returned vector for that group.
 - A data frame, a named tuple of vectors or a matrix gives the same additional columns
   and as many rows for each group as the rows returned for that group.
+  In this case returning a table with no columns drops group.
```

Comment:
Maybe just say `"(a table with zero rows drops the group)"`? That's just a natural extension of the general rule given by the previous sentence, and the presence of columns or not doesn't change the result as long as there are no rows.

Reply:
OK. Fixed (but "columns" not "rows" 😄 )

Reply:
I really meant "rows". A column with zero columns also has zero rows, and all tables with zero rows give the same result, right?

Reply:
There is a subtle thing:

a table with zero rows must have zero columns or have column names matching `colnames`. If table has zero rows but non-zero columns that have wrong names an error will be thrown (and this is intended).

What would you then suggest to write to convey this message?

Reply:
OK. Maybe "As a special case, returning an empty table with zero rows and columns is allowed, whatever the number of columns returned for other groups."?

Reply:
Actually we allow non-zero rows as long as it is zero columns, e.g. in `rand(10, 0)`, at least now. We could disallow it, but it would introduce another check that would only slow things down a bit, for no practical benefit.

I will write "As a special case, returning an empty table with zero columns is allowed, whatever the number of columns returned for other groups." as `isempty` and "zero columns" are correct conditions (they are duplicate but I think it is OK to give both as it will be easier to understand it by users).